### ISSUE TITLE
CFP: Node label context options

### ISSUE THREAD

**Issue by brancz** [cilium/cilium#33839]
## Cilium Feature Proposal
        
Thanks for taking time to make a feature proposal for Cilium! If you have usage questions, please try the [slack channel](http://slack.cilium.io/) and see the [FAQ](https://github.com/cilium/cilium/issues?utf8=%E2%9C%93&q=is:issue+label:kind/question+) first.

**Is your proposed feature related to a problem?**

Ultimately what I would like to do is be able to monitor how much traffic is crossing zone boundaries within a cloud provider region. There are two components to make that happen:

1) Traffic bandwidth analysis (covered by https://github.com/cilium/cilium/issues/16188)
2) The ability to tell src and destination zone.

**Describe the feature you'd like**

Node labels to be available in context options: https://docs.cilium.io/en/stable/observability/metrics/#context-options

**(Optional) Describe your proposed solution**

tl;dr we already find the pod name for the src and dst, so just extend this to the node name they are scheduled on and read the labels from those nodes.

Please complete this section if you have ideas / suggestions on how to implement the feature. We strongly recommend discussing your approach with Cilium committers before spending lots of time implementing a change. 

For longer proposals, you are welcome to link to an external doc (e.g. a Google doc). We have a [Cilium Feature Proposal template](https://docs.google.com/document/d/1vtE82JExQHw8_-pX2Uhq5acN1BMPxNlS6cMQUezRTWg/edit) to help you structure your proposal - if you would like to use it, please make a copy and ensure it's publicly visible, and then add the link here.

Once the CFP is close to being finalized, please add it as a PR to the [design-cfps](https://github.com/cilium/design-cfps) repo for final approval.

**Comment by brancz** [comment]
Looks like my prayers were heard https://github.com/cilium/cilium/pull/32851

**Comment by chancez** [comment]
For this I think you'll still need destination node labels which we don't have in flows, only source node labels. See https://github.com/cilium/cilium/issues/33601 / https://github.com/cilium/cilium/issues/34133.
